UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs

```diff
@@ -729,21 +728,36 @@ AnimationClip ConstructClip(Transform root, Transform[] nodes, int animationId)
 						break;
 
 					case GLTFAnimationChannelPath.scale:
-
 						for (var i = 0; i < input.AsFloats.Length; ++i)
 						{
 							var time = input.AsFloats[i];
-							var position = output.AsVec3s[i];
-							curveX.AddKey(time, position.X);
-							curveY.AddKey(time, position.Y);
-							curveZ.AddKey(time, position.Z);
+							Vector3 scale = output.AsVec3s[i].ToUnityVector3Raw();
+							curveX.AddKey(time, scale.x);
+							curveY.AddKey(time, scale.y);
+							curveZ.AddKey(time, scale.z);
 						}
 
 						clip.SetCurve(relativePath, typeof(Transform), "localScale.x", curveX);
 						clip.SetCurve(relativePath, typeof(Transform), "localScale.y", curveY);
 						clip.SetCurve(relativePath, typeof(Transform), "localScale.z", curveZ);
+						break;
+
+					case GLTFAnimationChannelPath.weights:
+						var primitives = channel.Target.Node.Value.Mesh.Value.Primitives;
+						var targetCount = primitives[0].Targets.Count;
+						for (int primitiveIndex = 0; primitiveIndex < primitives.Count; primitiveIndex++)
+						{
+							string primitiveObjPath = relativePath + "/Primitive" + primitiveIndex;
+							for (int targetIndex = 0; targetIndex < targetCount; targetIndex++)
+							{
+								//clip.SetCurve(primitiveObjPath, typeof(SkinnedMeshRenderer), "blendShape." + targetIndex, curves[targetIndex]);
```

Comment:
remove, leave comment, or reactivate

Reply:
I put a TODO to add blend shapes/morph targets. I 'm going to do that in a separate PR since blendshapes have more than 4 curves and will need some re-jiggering of the animation construction code